webapp/tests/karma/unit/services/db.js

```diff
@@ -45,7 +45,7 @@ describe('DB service', () => {
       getService = () => {
         // delay initialisation of the db service
         const service = $injector.get('DB');
-        $timeout.flush();
+        $timeout.flush(1000);
```

Comment:
Is this necessary?

Reply:
Apparently yes but I have no idea why. This started throwing an error on my branch for one specific test (the last one in this file). Turns out `$timeout.flush()` will throw an error if there's nothing to flush, which is the case for the last test due to [this clause](https://github.com/medic/medic-webapp/blob/master/webapp/src/js/services/db.js#L69), and adding a corresponding delay of 1000 prevents that error. For some reason the changes I introduced have somehow influenced this behaviour but I cannot see why or how, so am a bit stumped.